### PR TITLE
added MemoryBeanProcessor

### PR DESCRIPTION
**Pull request by OlehSarapuk**

### CODE REVIEW

**Comment by OlehSarapuk** [comment]
@ajax-gnatiuk-s please review

**Comment by ajax-gnatiuk-s** [comment]
@ArSiu
Review please

File: src/main/java/com/example/ajaxonboardingproject/config/MemoryBeanPostProccesor.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+
+const val MAX_MEMORY_USAGE: Long = 1
+
+@Component
+@Suppress("SpreadOperator")
+class MemoryCheckBeanPostProcessor : BeanPostProcessor {
+    val map: MutableMap<String, Class<*>> = mutableMapOf()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/com/example/ajaxonboardingproject/config/MemoryBeanPostProccesor.kt]: should be `private`?

File: src/main/java/com/example/ajaxonboardingproject/config/MemoryBeanPostProccesor.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any? {
+        val originalBeanClass = map[beanName]
+        if (originalBeanClass != null) {
+            val enhancer = Enhancer()
+            enhancer.setSuperclass(originalBeanClass)
+            enhancer.setCallback(MethodInterceptor { _, method, args, _ ->
+                val before = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
+                val result = method.invoke(bean, *args)
+                val after = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
```
Comment (ajax-gnatiuk-s) [inline src/main/java/com/example/ajaxonboardingproject/config/MemoryBeanPostProccesor.kt]: what about extracting a function:
```kotlin
private fun computeUsedMemory() = 
                  Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()

```

**Review by ArSiu — APPROVED** [review]

**Review by ajax-gnatiuk-s — APPROVED** [review]

**Pull request merged**